.github/workflows/go.yml

```diff
@@ -1,125 +1,25 @@
-        run: find . -name go.mod -not -path "*wrappers/transit/*" -execdir sh -c 'go test ./... || exit 1' sh {} +
-
-  sqlite:
-    strategy:
-      matrix:
-        go: ["1.21", "1.20"]
-        platform: [ubuntu-latest] # can not run in windows OS
-    runs-on: ${{ matrix.platform }}
-
-    steps:
-      - name: Set up Go 1.x
-        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
-        with:
-          go-version: ${{ matrix.go }}
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
-
-      - name: go mod package cache
-        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
-        with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ matrix.go }}-${{ hashFiles('tests/go.mod') }}
-
-      - name: Tests
-        working-directory: ./extras/kms
-        run: make test-sqlite
-
-  postgres:
-    strategy:
-      matrix:
-        dbversion:
-          [
-            "postgres:14",
-            "postgres:13",
-            "postgres:12",
-            "postgres:11",
-            "postgres:10",
-          ]
-        go: ["1.21", "1.20"]
-        platform: [ubuntu-latest] # can not run in macOS and Windows
-    runs-on: ${{ matrix.platform }}
-
-    services:
-      postgres:
-        image: ${{ matrix.dbversion }}
-        env:
-          POSTGRES_PASSWORD: go_db
-          POSTGRES_USER: go_db
-          POSTGRES_DB: go_db
-        ports:
-          - 9920:5432
-        # Set health checks to wait until postgres has started
-        options: >-
-          --health-cmd pg_isready
-          --health-interval 10s
-          --health-timeout 5s
-          --health-retries 5
-
-    steps:
-      - name: Set up Go 1.x
-        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
-        with:
-          go-version: ${{ matrix.go }}
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
-
-      - name: go mod package cache
-        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
-        with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ matrix.go }}-${{ hashFiles('tests/go.mod') }}
-
-      - name: Tests
-        working-directory: ./extras/kms
-        run: make test-postgres
-
-  plugin-transit:
-    strategy:
-      matrix:
-        go: ["1.21", "1.20"]
+        directory:
+          - '.'
+          - './entropy'
+          - './plugin'
+          - './wrappers/aead'
+          - './wrappers/alicloudkms'
+          - './wrappers/awskms'
+          - './wrappers/azurekeyvault'
+          - './wrappers/gcpckms'
+          - './wrappers/huaweicloudkms'
+          - './wrappers/ocikms'
+          - './wrappers/tencentcloudkms'
+          - './wrappers/transit'
+        go: ["1.22", "1.21"]
```

Comment:
Does it even make sense to keep older Go versions?

Reply:
I think just keeping the one we use for the main repo is fine

Reply:
Thanks @JanMa, updated!